src/city/health.c

```diff
@@ -19,8 +20,113 @@ void city_health_change(int amount)
     city_data.health.value = calc_bound(city_data.health.value + amount, 0, 100);
 }
 
+static void cause_plague_in_building(int building_id)
+{
+    building *b = building_get(building_id);
+    if (!b->ruin_has_plague) {
+
+        // remove all eatable resources from granary and warehouse
+        if (b->type == BUILDING_GRANARY) {
+            for (int resource = RESOURCE_MIN_FOOD; resource < RESOURCE_MAX_FOOD; resource++) {
+                building_granary_remove_resource(b, resource, 3200);
+            }
+        }
+
+        if (b->type == BUILDING_WAREHOUSE) {
+            for (int resource = RESOURCE_MIN_FOOD; resource < RESOURCE_IRON; resource++) {
+                building_warehouse_remove_resource(b, resource, 32);
+            }
+        }
+
+        // set building to plague status and use fire process to manage plague on it
+        b->ruin_has_plague = 1;
+        b->fire_proof = 1;
```

Comment:
I was to prevent fire in case of plague because I didn't know how figures works. But yes we can remove this fire_proof = 1 because in case of fire, b->figure_4 is set to 0